src/main/java/org/jabref/logic/importer/fetcher/MergingIdBasedFetcher.java

```diff
@@ -0,0 +1,92 @@
+            List.of(StandardField.DOI, StandardField.ISBN, StandardField.EPRINT);
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MergingIdBasedFetcher.class);
+    private final ImportFormatPreferences importFormatPreferences;
+
+    public MergingIdBasedFetcher(ImportFormatPreferences importFormatPreferences) {
+        this.importFormatPreferences = importFormatPreferences;
+    }
+
+    public Optional<FetcherResult> fetchEntry(BibEntry entry) {
+        LOGGER.debug("Entry {}", entry);
+
+        return SUPPORTED_FIELDS.stream()
+                               .map(field -> tryFetch(entry, field))
+                               .filter(Optional::isPresent)
+                               .map(Optional::get)
+                               .findFirst();
+    }
+
+    private Optional<FetcherResult> tryFetch(BibEntry entry, Field field) {
+        return entry.getField(field)
+                    .flatMap(identifier -> getFetcherForField(field)
+                            .flatMap(fetcher -> fetchEntry(fetcher, field, identifier, entry)));
+    }
+
+    private Optional<FetcherResult> fetchEntry(IdBasedFetcher fetcher, Field field,
+                                               String identifier, BibEntry entryFromLibrary) {
+        try {
+            LOGGER.debug("Entry {}",
+                    entryFromLibrary);
+
+            return fetcher.performSearchById(identifier)
+                          .map(fetchedEntry -> {
+                              BibEntry mergedEntry = (BibEntry) entryFromLibrary.clone();
+                              boolean hasChanges = mergeBibEntry(fetchedEntry, mergedEntry);
+                              return new FetcherResult(entryFromLibrary, mergedEntry, hasChanges);
+                          });
+        } catch (Exception exception) {
+            LOGGER.error("Error fetching entry with {} identifier: {}",
+                    field, identifier, exception);
+            return Optional.empty();
+        }
+    }
+
+    private boolean mergeBibEntry(BibEntry source, BibEntry target) {
+        boolean hasChanges = false;
+        for (Field field : source.getFields()) {
+            Optional<String> sourceValue = source.getField(field);
+            Optional<String> targetValue = target.getField(field);
+
+            if (sourceValue.isPresent() && targetValue.isEmpty()) {
+                target.setField(field, sourceValue.get());
+                hasChanges = true;
+            }
+        }
+        return hasChanges;
+    }
+
+    private Optional<IdBasedFetcher> getFetcherForField(Field field) {
+        return switch (field) {
+            case StandardField.DOI -> Optional.of(new DoiFetcher(importFormatPreferences));
+            case StandardField.ISBN -> Optional.of(new IsbnFetcher(importFormatPreferences));
+            case StandardField.EPRINT -> Optional.of(new IacrEprintFetcher(importFormatPreferences));
+            default -> Optional.empty();
+        };
+    }
```

Comment:
Why this? Above, at the refactoring, you called `org.jabref.logic.importer.WebFetchers#getIdBasedFetcherForField`, which offers the same functionality.

Did you use AI for this and did not get that this method already exists?

Reply:
The implementation of `getFetcherForField` was actually manually added by me as an initial step following your suggestion to "factor out `WebFetchers.getIdBasedFetcherForField`" from `FetchAndMergeEntry#fetchAndMerge`.
I was actually confused for creating duplicate functionality at that time. Would you prefer that I directly use `WebFetchers.getIdBasedFetcherForField` instead of using this separate implementation?

Reply:
"factor" out also means to re-use things. I have no time now to search for my comment. Do you know that GitHub offers deep links to comments? Maybe, next time when back referencing, use a link to the concrete comment. That helps. Reason: I have so many pull requests to comment on - and also other things todo, I cannot remember the complete code base and the complete pull request.

Please try to use existing code. It seems you need to remove ` getFetcherForField` and replace calls to it by `WebFetchers.getIdBasedFetcherForField`. This is not too hard, is it?

I do not know the AI commands for this and it seems that also the AI cannot have the whole code base in its "mind".

Reply:
Thank you for your clarification. Sorry for wasting your time. I understand you are quite busy with so many Pull Requests at the same time. I will use deep links next time for your convenience if necessary.
I will invoke the existing method and delete the redundant code. And then I will try to write some unit tests for the new classes. However, I am currently dealing with my final exam weeks, I would be busy until November 14th . Is it OK for me to continue the work after that? Thank you for your understanding.

Reply:
Sure, take your time!